content/en/apps/guides/data/analytics/building-dbt-models.md

```diff
@@ -0,0 +1,340 @@
+    ]
+  )
+}}
+
+SELECT
+  chw.uuid as chw_uuid,
+  clinic.uuid as clinic_uuid,
+  health_center.uuid as health_center_uuid,
+  district_hospital.uuid as district_hospital_uuid
+FROM
+  {{ref('contact')}} chw
+  INNER JOIN {{ref('contact')}} clinic ON chw.parent_uuid = clinic.uuid
+  LEFT JOIN {{ref('contact')}} health_center ON clinic.parent_uuid = health_center.uuid
+  LEFT JOIN {{ref('contact')}} district_hospital ON health_center.parent_uuid = district_hospital.uuid
+WHERE chw.contact_type = 'person' AND clinic.contact_type = 'clinic';
+```
+
+### Aggregates
+
+To aggregate on contacts and report data, join to the `data_record` and `contacts` tables as in the example below.
+
+```sql
+SELECT
+  COUNT(*),
+  chw.uuid as chw_uuid,
+  chw.name as chw_name,
+  area.uuid as area_uuid,
+  area.name as area_name,
+  date_trunc('month', pregnancy.reported) as report_month
+FROM
+  pregnancy
+INNER JOIN contacts chw ON chw.uuid = data_record.contact_uuid
+LEFT JOIN contacts area ON area.uuid = data_record.parent_contact_uuid
+GROUP BY
+  chw.uuid,
+  chw.name,
+  area.uuid,
+  area.name,
+  report_month;
+```
+
+### Aggregations
+It may be useful to save some common queries as views, so that they can be reused in dashboards or downstream aggregations.
+With the pregnancy form model above, if there was also a postnatal care form they could be joined to view outcomes together with registrations.
+
+```sql
+{{
+    config(materialized = 'materialized_view')
+}}
+
+SELECT
+  pnc.uuid AS pnc_uuid,
+  pregnancy.uuid AS pregnancy_uuid,
+
+  pnc.patient_id AS patient_id,
+
+  pnc_contact.uuid AS pnc_reported_by,
+  pregnancy_contact.uuid AS pregnancy_reported_by,
+  pnc.reported AS reported,
+  pregnancy.reported AS pregnancy_reported,
+
+  pregnancy.edd as edd,
+  pnc.delivery_date AS delivery_date,
+  pnc.pregnancy_outcome,
+FROM
+  {{ ref('postnatal_care') }} AS pnc
+LEFT JOIN {{ ref('pregnancy') }} ON
+  (pnc.reported > pregnancy.reported AND
+   pnc.reported < (pregnancy.reported + '1 year'::INTERVAL) AND
+   pnc.patient_id = pregnancy.patient_id)
+WHERE
+  (pnc.pregnancy_outcome IN (VALUES ('healthy'), ('still_birth')))
+  AND pnc.patient_id IS NOT NULL AND pnc.patient_id <> ''
+```
+
+## Views vs. Incremental Tables
+
+There are two options for building models using DBT: _views_ and _incremental tables_; see a full description [in the DBT documentation](https://docs.getdbt.com/docs/build/materializations).
+In short, views are normal SQL views, and incremental tables are actual tables, which are periodically updated as new data is created.
+
+For CHT DBT models, it is generally preferred to start with views. Views are always up to date and do not require a full refresh when changed.
+However, indexing the columns of views is impossible, and views built on top of other views can cause performance issues.
+
+To convert a view to an incremental table, change the materialization to `incremental` and add a condition to the `WHERE` clause:
+
```

Comment:
We should mention here that it is preferable to have model being queried by the dashboard to be materialized views and not tables to avoid having them unavailable when DBT is refreshing them.